test/Microsoft.AspNet.Http.Abstractions.Tests/UseMiddlewareTest.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Builder.Internal;
+using Xunit;
+
+namespace Microsoft.AspNet.Http
+{
+    public class UseMiddlewareTest
+    {
+        [Fact]
+        public void UseMiddlewareNoParameters()
+        {
+            var mockServiceProvider = new MockServiceProvider();
+            var builder = new ApplicationBuilder(mockServiceProvider);
+            builder.UseMiddleware(typeof(MiddlewareNoParameters));
+            var exception = Assert.Throws<InvalidOperationException>(() => builder.Build());
+            Assert.Equal("Middleware Invoke method must take first argument of HttpContext", exception.Message); 
+        }
+
+        [Fact]
+        public void UseMiddlewareNonTaskReturnType()
+        {
+            var mockServiceProvider = new MockServiceProvider();
+            var builder = new ApplicationBuilder(mockServiceProvider);
+            builder.UseMiddleware(typeof(MiddlewareNonTaskReturn));
+            var exception = Assert.Throws<InvalidOperationException>(() => builder.Build());
+            Assert.Equal("Invoke does not return an object of type Task", exception.Message);
+        }
+
+        [Fact]
+        public void UseMiddlewareInvokeMissing()
+        {          
+            var mockServiceProvider = new MockServiceProvider();
+            var builder = new ApplicationBuilder(mockServiceProvider);
+            builder.UseMiddleware(typeof(MiddlewareNoInvoke));
+            var exception = Assert.Throws<InvalidOperationException>(() => builder.Build());
+            Assert.Equal("No public Invoke method found", exception.Message);
+        }
+
+        private class MockServiceProvider : IServiceProvider
```

Comment:
After looking at the article I'd agree with classifying it as a Dummy class. 
I changed the test middleware classes to be "stubs" since they can't really respond "at all to anything outside what's programmed in for the test". What do you think?